### PR TITLE
PRT header

### PR DESCRIPTION
**Pull request by nairr**
Adds header for .PRT file with the following information
1. Flow binary information
2. host names(s)/ machine names
3. host hardware
4. Operating system
5. User
6. System time and date
prtheader

### CODE REVIEW

**Comment by atgeirr** [comment]
jenkins run spe norne_ebos please

File: opm/autodiff/FlowMainEbos.hpp [hunk 1]
```diff
@@ -23,6 +23,12 @@
 #ifndef OPM_FLOW_MAIN_EBOS_HEADER_INCLUDED
 #define OPM_FLOW_MAIN_EBOS_HEADER_INCLUDED
 
+#include <sys/utsname.h>
+#include <unistd.h>
+#include <ctime>
+#include <sys/statvfs.h>
+#include <sys/sysinfo.h>
```
Comment (atgeirr) [inline opm/autodiff/FlowMainEbos.hpp]: Unfortunately this header seems to be a linux-specific thing. It is only used to extract the RAM size of the system. The simplest solution is to simply drop that particular output, but if you think it is very important I can try to find a cross-platform solution.

So: is it necessary to have RAM output?

File: opm/autodiff/FlowMainEbos.hpp [hunk 2]
```diff
@@ -189,7 +196,8 @@ namespace Opm
 
         // Print startup message if on output rank.
         void printStartupMessage()
-        {
+        {
```
Comment (atgeirr) [inline opm/autodiff/FlowMainEbos.hpp]: Extra spaces and blank line here cause a difference, please remove.

File: opm/autodiff/FlowMainEbos.hpp [hunk 3]
```diff
@@ -318,13 +326,56 @@ namespace Opm
             streamLog->setMessageLimiter(std::make_shared<MessageLimiter>(10, limits));
             streamLog->setMessageFormatter(std::make_shared<SimpleMessageFormatter>(true));
 
-            // Read parameters.
             if ( output_cout_ )
-            {
-                OpmLog::debug("\n---------------    Reading parameters     ---------------\n");
+            { 
+            // Read Parameters.
```
Comment (atgeirr) [inline opm/autodiff/FlowMainEbos.hpp]: Please indent correctly.

File: opm/autodiff/FlowMainEbos.hpp [hunk 4]
```diff
@@ -318,13 +326,56 @@ namespace Opm
...
+            OpmLog::debug("\n---------------    Reading parameters     ---------------\n");
             }
         }
-
+        
+        void printPRTHeader()
+        {
+            if ( output_cout_ )
+            {
```
Comment (atgeirr) [inline opm/autodiff/FlowMainEbos.hpp]: Please place brace at the end of the if-line and indent the code block.

File: opm/autodiff/FlowMainEbos.hpp (hunk 1 above)
Comment (akva2) [inline opm/autodiff/FlowMainEbos.hpp]:
```C
#include <unistd.h>

unsigned long long getTotalSystemMemory()
{
    long pages = sysconf(_SC_PHYS_PAGES);
    long page_size = sysconf(_SC_PAGE_SIZE);
    return pages * page_size;
}
```

should work on osx as well.

File: opm/autodiff/FlowMainEbos.hpp [hunk 5]
```diff
@@ -318,13 +326,56 @@ namespace Opm
...
+        void printPRTHeader()
+        {
+            if ( output_cout_ )
+            { 
+            // Print header for PRT file.      
+            const std::string version = moduleVersionName();
+            const double megabyte = 1024 * 1024;
+            unsigned num_cpu = std::thread::hardware_concurrency();    
+            struct utsname arch;
+            char userName[64] = {0};
```
Comment (atgeirr) [inline opm/autodiff/FlowMainEbos.hpp]: This part is confusing: userName is never used. I think it would be better to do it like this (replacing the current lines 370-373):
```
const char* user = getlogin();
if (user) {
    ss << "User          =  " << user << std::endl;
}
```

File: opm/autodiff/FlowMainEbos.hpp [hunk 6]
```diff
@@ -792,4 +843,4 @@ namespace Opm
     };
 } // namespace Opm
 
-#endif // OPM_FLOW_MAIN_EBOS_HEADER_INCLUDED
+#endif // OPM_FLOW_MAIN_EBOS_HEADER_INCLUDED
```
Comment (atgeirr) [inline opm/autodiff/FlowMainEbos.hpp]: Unnecessary change.

**Comment by atgeirr** [comment]
Overall this is nice! I have commented on some minor issues above, and one major (cross-platform printout of total RAM). Please tell me if the RAM is important, if so I'll help find a solution.

Other than that, this will be merged when issues are resolved, thanks!

**Comment by akva2** [comment]
i already suggested a fix for the memory thing above.

File: opm/autodiff/FlowMainEbos.hpp (hunk 1 above)
Comment (atgeirr) [inline opm/autodiff/FlowMainEbos.hpp]: I can confirm that it does, thanks!

**Comment by atgeirr** [comment]
jenkins run norne please

**Comment by atgeirr** [comment]
jenkins run norne_ebos please

**Comment by akva2** [comment]
said the goldfish.

**Comment by akva2** [comment]
jenkins run norne please

**Comment by atgeirr** [comment]
> said the goldfish.

I checked the console log and it clearly ran flow_legacy (while I wanted to test flow_ebos).

**Comment by akva2** [comment]
fixed now. forgot to update the opm-simulators builder to use the now-shared script.

**Comment by akva2** [comment]
stupid pr builder bug. please no comments until it triggers :/

**Comment by akva2** [comment]
jenkins run norne please

**Comment by atgeirr** [comment]
Jenkins is green and issues resolved, merging!
